mime.go

```diff
@@ -24,6 +25,16 @@ func DetectReader(r io.Reader) (mime, extension string, err error) {
 	in = in[:n]
 
 	mime, ext := Detect(in)
+
+	if err == nil { //file size is more than 520 bytes
```

Comment:
Hi @gabriel-vasile ,
sorry for the delayed response and thanks for reviewing the changes :) !
The concerned raised is, if extension is not matched at all, then it will try again for FullDataNodesMap.
But I think, it will not do so. 
FullDataNodesMap is the map of nodes which needs full data to perform the sniffing. 
For now, FullDataNodesMap is only having "custom Txt node with json as only child node" as the only key present. So, json check will happen only if the line 27 responds back with Txt as the mimetype. If Root node is returned as output by the line 27 , then the node will not be found in the map and hence check will not be performed. The only overhead, in Root node case, will be the key lookup of FullDataNodesMap map which i think will be relatively small and can be safely ignored (as not many keys will be present).
Also if any of the child node is identified correctly or incorrectly by line 27, it will be looked up again only if the node is mentioned in the map. To quote the example in comment, if html is matched , then the mimetype will be html and hence the map lookup for html will fail (it is having onlty Txt as the mime type key for now). So if html is passed then json lookup wont happen again.
To rephrase your question, the performance overhead will come into picture when line 27 returns Txt as node (not Root or Html). To be more specific , if any of the type present in the map is returned then only the complete file check will be triggered. 
There will be two cases in such lookup : 
1 Txt was identified because none of the child nodes were matched.
2 Txt was correctly identified  

In first case, matching is required as the given file may be json.
In Second case, triggering of the node-matching again is inefficient/redundant. 
But in this case also, only json matching(only one node) is performed. Note that the Txt is only having Json as single child node unlike the original Txt node. 
Was this the second case you were referring to in your comment ?

Thanks Again And Regards.

Reply:
I think I understood how the code works now and if I'm not missing anything, then at line#33 the matching was supposed be done on `rootNode` instead of `Root`. Regardless of that, I'm not too keen on this approach of reading the whole file in memory to detect it. On top of that the files are stored twice in memory: once they are stored in `remainingData` (missing first 520 bytes), and secondly in `allData`. Even for few MB files, this memory usage is too much. I think we need another solution for this problem with json detected as txt.